controllers/rack.go

```diff
@@ -702,31 +685,46 @@ func (r *SingleClusterReconciler) rollingRestartRack(
 			" config",
 	)
 
-	for _, pod := range podList {
+	// Find pods which needs restart
+	var podsToRestart []*corev1.Pod
+	for i := range podList {
+		pod := podList[i]
+
 		// Check if this pod needs restart
 		restartType, err := r.getRollingRestartTypePod(rackState, pod)
 		if err != nil {
 			return found, reconcileError(err)
 		}
 
 		if restartType == NoRestart {
-			r.Log.Info(
-				"This Pod doesn't need rolling restart, Skip this", "pod",
-				pod.Name,
-			)
+			r.Log.Info("This Pod doesn't need rolling restart, Skip this", "pod", pod.Name)
 			continue
 		}
+		podsToRestart = append(podsToRestart, &pod)
+	}
 
-		res := r.rollingRestartPod(rackState, pod, restartType, ignorablePods)
+	// Failed pods should always come before active pods.
+	// We make info calls before restarting active pods,
+	// if active pods are checked before failed pods then
+	// info call will fail in failed pods and operator will not go ahead
+	rearrangedPods := getRearrangedFailedAndActivePods(podsToRestart)
+
+	// Create batch of pods
+	podsBatchList := r.getPodsBatchToRestart(rearrangedPods, len(podList))
+
+	// Restart batch of pods
+	for _, podsBatch := range podsBatchList {
```

Comment:
this for loop is never going to hit more than once as, it has returned in every scenario.